packages/amplify-provider-awscloudformation/src/resource-push/ResourceDeployer.ts

```diff
@@ -0,0 +1,329 @@
+          ...resource,
+          lastBuildTimeStamp,
+        };
+      }),
+    );
+  }
+
+  /**
+   * zips buildable project and saves it dist
+   * @param builtResources {BuiltResourceDefinition[]}
+   * @returns resources with params for packaging {PackagedResourceDefinition[]}
+   */
+  protected async packageResources(builtResources: BuiltResourceDefinition[]): Promise<PackagedResourceDefinition[]> {
+    const { FUNCTION_CATEGORY } = Constants;
+    return await Promise.all(
+      builtResources.map(async resource => {
+        if (!resource.build) {
+          return resource;
+        }
+        const result: PackagerParams = await this.context.amplify.invokePluginMethod(
+          this.context,
+          FUNCTION_CATEGORY.NAME,
+          resource.service,
+          'packageResource',
+          [this.context, resource, true],
+        );
+        return {
+          ...resource,
+          packagerParams: result,
+        };
+      }),
+    );
+  }
+  /**
+   * Changes that need to be made after packaging takes place
+   * @param pacakgedResources generated by {packageResource}
+   * @returns {PackagedResourceDefinition[]}
+   */
+  protected async postPackageResource(pacakgedResources: PackagedResourceDefinition[]): Promise<PackagedResourceDefinition[]> {
+    const { options } = this.context.parameters;
+    const { API_CATEGORY } = Constants;
+    if (this.resourcesHasCategoryService(pacakgedResources, API_CATEGORY.NAME, API_CATEGORY.SERVICE.APP_SYNC)) {
+      await transformGraphQLSchema(this.context, {
+        handleMigration: opts => updateStackForAPIMigration(this.context, 'api', undefined, opts),
+        minify: options['minify'],
+      });
+    }
+    return pacakgedResources;
+  }
+
+  /**
+   * Checks if the project has Containers API resource
+   * @param packagedResources
+   * @returns {boolean} true of false
+   */
+  protected resourcesHasContainers(packagedResources: PackagedResourceDefinition[]): boolean {
+    const { API_CATEGORY, HOSTING_CATEGORY } = Constants;
+    return (
+      this.resourcesHasCategoryService(packagedResources, API_CATEGORY.NAME, API_CATEGORY.SERVICE.ELASTIC_CONTAINER) ||
+      this.resourcesHasCategoryService(packagedResources, HOSTING_CATEGORY.NAME, HOSTING_CATEGORY.SERVICE.ELASTIC_CONTAINER)
+    );
+  }
+
+  /**
+   * Saves providerMetadata s3 information
+   * @param packagedResource
+   * @param bucketName
+   */
+  protected storeS3BucketInfo(packagedResource: PackagedResourceDefinition, bucketName: string): void {
+    const s3Info = {
+      deploymentBucketName: bucketName,
+      s3Key: packagedResource.packagerParams.zipFilename,
+    };
+    const { CATEGORIES, S3_BUCKET } = Constants;
+    _.set(
+      this.amplifyTeamProviderInfo,
+      [this.envInfo.envName, CATEGORIES, packagedResource.category, packagedResource.resourceName],
+      s3Info,
+    );
+    _.set(this.amplifyMeta, [packagedResource.category, packagedResource.resourceName, S3_BUCKET], s3Info);
```

Comment:
Does this mean the uploading of resources happen in CLI and we are not using CDK to upload?

Reply:
This mutates the amplifyMeta in memory and doesn't commit it to the file. I had to do this so because `formNested` function stack looks at these value to generate the root stack.